File: src/python/test/TestEither.py
import unittest
from Utils import Utils 
from Refi import Refi



class TestEither(unittest.TestCase):

    def setUp(self):
        self.refi = Refi()
        self.utils = Utils()


    def test_Either(self):


        expression = self.refi.toRegEx("`a` either{`b`, `c`} `d`")
        self.utils.assertRegexIs("a[bc]d", expression)
        self.utils.assertRegexMatchesText(expression, "abd")
        self.utils.assertRegexMatchesText(expression, "acd")
        self.utils.assertRegexDoesntMatchText(expression, "ad")
        self.utils.assertRegexDoesntMatchText(expression, "amd")      
    
    def test_EitherComplex(self):

        expression = self.refi.toRegEx("`a` either{`b`, `c`, `d`} `e`")
        self.utils.assertRegexIs("a[bcd]e", expression)
        self.utils.assertRegexMatchesText(expression, "abe")
        self.utils.assertRegexMatchesText(expression, "ace")
        self.utils.assertRegexMatchesText(expression, "ade")
        self.utils.assertRegexDoesntMatchText(expression, "ad")
        self.utils.assertRegexDoesntMatchText(expression, "amd")      


    def runTest(self):
        pass  

if __name__ == '__main__':
    unittest.main()    
    

File: src/test/java/koorkah/TestEither.java
package koorkah;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

class TestEither extends BaseT
{

    @Test
    void testEither()
    {

        String expression = koorkah.toRegEx("`a` either{`b`, `c`} `d`");
        assertEquals("a[bc]d", expression);
        assertRegexMatchesText(expression, "abd");
        assertRegexMatchesText(expression, "acd");
        assertRegexDoesntMatchText(expression, "ad");
        assertRegexDoesntMatchText(expression, "amd");      
    }
    
    @Test
    void testEitherComplex()
    {
        String expression = koorkah.toRegEx("`a` either{`b`, `c`, `d`} `e`");
        assertEquals("a[bcd]e", expression);
        assertRegexMatchesText(expression, "abe");
        assertRegexMatchesText(expression, "ace");
        assertRegexMatchesText(expression, "ade");
        assertRegexDoesntMatchText(expression, "ad");
        assertRegexDoesntMatchText(expression, "amd");      
    }


}
